enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/vault/SecretNode.java

```diff
@@ -18,36 +18,162 @@
+import com.oracle.bmc.vault.responses.GetSecretResponse;
+import com.oracle.bmc.vault.responses.ScheduleSecretDeletionResponse;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
 import java.util.stream.Collectors;
+import javax.swing.Action;
 import org.netbeans.modules.cloud.oracle.ChildrenProvider;
 import org.netbeans.modules.cloud.oracle.NodeProvider;
-import static org.netbeans.modules.cloud.oracle.OCIManager.getDefault;
+import static org.netbeans.modules.cloud.oracle.NotificationUtils.confirmAction;
+import static org.netbeans.modules.cloud.oracle.NotificationUtils.showErrorMessage;
+import static org.netbeans.modules.cloud.oracle.NotificationUtils.showMessage;
+import org.netbeans.modules.cloud.oracle.OCIManager;
 import org.netbeans.modules.cloud.oracle.OCINode;
 import org.netbeans.modules.cloud.oracle.items.OCID;
+import org.netbeans.modules.cloud.oracle.items.OCIItem;
+import org.openide.actions.DeleteAction;
 import org.openide.nodes.Children;
+import org.openide.util.NbBundle;
+import org.openide.util.RequestProcessor;
+import org.openide.util.actions.SystemAction;
 
 /**
  *
  * @author Jan Horvath
  */
+@NbBundle.Messages({
+    "SecretNodeDesc=Valut Secret: {0}\nLifecycle State: {1}{2}",
+    "# {0} - [OCIItem name]",
+    "MSG_ConfirmDeleteAction=Are you sure that you want to schedule deletion of {0}",
+    "# {0} - [OCIItem name]",
+    "MSG_DeleteActionFailed=Failed to schedule deletion of {0}.",
+    "# {0} - [OCIItem name]", "# {1} - [Scheduled deletion time]",
+    "MSG_DeleteActionSuccess=Successfully scheduled deletion of {0} at {1}."
+})
 public class SecretNode extends OCINode {
     private static final String SECRET_ICON = "org/netbeans/modules/cloud/oracle/resources/secret.svg"; // NOI18N
+    private static final SimpleDateFormat DELETION_TIME_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
 
     public SecretNode(SecretItem vault) {
         super(vault, Children.LEAF);
         setName(vault.getName());
         setDisplayName(vault.getName());
         setIconBaseWithExtension(SECRET_ICON);
-        setShortDescription(vault.getDescription());
+        setShortDescription(
+                createShortDescription(
+                        vault.getLifecycleState(),
+                        vault.getDeletionTime()));
+    }
+    
+    private String createShortDescription(String state, Date deletionTime) {
+        if (deletionTime != null) {
+            return Bundle.SecretNodeDesc(this.getItem().getName(), state, getDeletionTimeInfo(deletionTime));
+        }
+        return Bundle.SecretNodeDesc(this.getItem().getName(), state, "");
+    }
+    
+    private String getDeletionTimeInfo(Date deletionTime) {
+        return "\nDeletion time: " + DELETION_TIME_FORMAT.format(deletionTime);
```

Comment:
This decoration ought to be in a bundle. Pls. create 2 bundle keys, one normal, the other for deleted items.